src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -213,29 +179,13 @@ public boolean publish(SlackRequest slackRequest) {
                     post = new HttpPost(url);
 
                 } else {
-                    url = "https://slack.com/api/" + apiEndpoint;
+                    url = "https://slack.com/api/" + slackRequest.getApiEndpoint();
 
                     post = new HttpPost(url);
                     post.setHeader("Authorization", "Bearer " + populatedToken);
-                    if (threadTs.length() > 1) {
-                        json.put("thread_ts", threadTs);
-                    }
-                    if (replyBroadcast) {
-                        json.put("reply_broadcast", "true");
-                    }
-                    if (StringUtils.isEmpty(iconEmoji) && StringUtils.isEmpty(username)) {
-                        json.put("as_user", "true");
-                    } else {
-                        if (StringUtils.isNotEmpty(iconEmoji)) {
-                            json.put("icon_emoji", iconEmoji);
-                        }
-                        if (StringUtils.isNotEmpty(username)) {
-                            json.put("username", username);
-                        }
-                    }
                 }
 
-                logger.fine("Posting: to " + roomId + " on " + teamDomain + " using " + url + ":  " + color);
+                logger.fine("Posting: to " + roomId + " on " + teamDomain + " using " + url + ". " + json.toString());
```

Comment:
`color` is no longer in scope (it lives in `SlackMessageRequest`). Having the full JSON of the request in the logs was useful when debugging. Since these are "fine" logs, it seems alright to me.